simulator/src/main/java/com/hedera/block/simulator/grpc/PublishStreamObserver.java

```diff
@@ -27,21 +30,31 @@
 public class PublishStreamObserver implements StreamObserver<PublishStreamResponse> {
 
     private final Logger logger = System.getLogger(getClass().getName());
+    private final AtomicBoolean streamEnabled;
 
-    /** Creates a new PublishStreamObserver instance. */
-    public PublishStreamObserver() {}
+    /** Creates a new PublishStreamObserver instance.
+     *
+     * @param streamEnabled is responsible for signaling, whether streaming should continue
+     */
+    public PublishStreamObserver(final AtomicBoolean streamEnabled) {
+        this.streamEnabled = streamEnabled;
```

Comment:
Maybe a `@NonNull` and a null check would be good here?

Reply:
Added. ✅